src/hdmf_zarr/zarr_utils.py

```diff
@@ -73,6 +74,18 @@ def __iter__(self):
     def __next__(self):
         return self._get_ref(super().__next__())
 
+    def append(self, arg):
+        # Create Builder
+        builder = self.invert().io.manager.build(arg)
+
+        # Get File/Source Object ID
+        f_builder = self.invert().io.read_builder()
+        source_file = self.invert().io.manager.construct(f_builder)
+
+        # Create ZarrReference
+        ref = self.invert().io._ZarrIO__get_ref(builder, source_file=source_file)
```

Comment:
Is there a difference between `self.invert().io` and `self.io`? I'm trying to follow the logic and am getting lost.